core/src/main/scala/shapeless/annotation.scala

```diff
@@ -45,15 +45,19 @@ trait Annotation[A, T] extends Serializable {
   def apply(): A
 }
 
-object Annotation {
+object Annotation extends Annotation0 {
+  implicit def materialize[A, T]: Annotation[A, T] = macro AnnotationMacros.materializeAnnotationRequired[A, T]
```

Comment:
I want to prioritize the default behavior not to decrease compilation speed, but it might be just okay to keep them in a single object.

Reply:
I don't think it matters for compilation speed.